x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/Matcher.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.execution.assembler;
+
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.xpack.eql.execution.search.Limit;
+import org.elasticsearch.xpack.eql.execution.search.Ordinal;
+import org.elasticsearch.xpack.eql.execution.sequence.Sequence;
+import org.elasticsearch.xpack.eql.execution.sequence.SequenceStateMachine;
+import org.elasticsearch.xpack.eql.session.Payload;
+
+import java.util.List;
+
+/**
+ * Executable tracking sequences at runtime.
+ */
+class Matcher {
```

Comment:
This used to be inside `SequenceRuntime` but now it has been extracted and takes care only of matching.